libflux/src/flux/lib.rs

```diff
@@ -286,24 +286,35 @@ pub unsafe extern "C" fn flux_merge_ast_pkgs(
 /// clauses match and merges the files from the input package into the output package. If
 /// package clauses fail validation then an option with an Error is returned.
 pub fn merge_packages(out_pkg: &mut ast::Package, in_pkg: &mut ast::Package) -> Option<Error> {
-    let pkg_clause = match &out_pkg.files[0].package {
-        Some(clause) => &clause.name.name,
-        None => return Some(Error::from("output package does not have a package clause")),
+    let out_pkg_clause = match &out_pkg.files[0].package {
+        Some(clause) => Some(&clause.name.name),
+        None => None,
     };
 
     // Check that all input files have a package clause that matches the output package.
     for file in &in_pkg.files {
         let file_clause = match &file.package {
-            Some(clause) => &clause.name.name,
-            None => return Some(Error::from("current file does not have a package clause")),
+            Some(clause) => Some(&clause.name.name),
+            None => None,
         };
 
-        if pkg_clause != file_clause {
-            return Some(Error::from(format!(
-                "file's package clause: {} does not match package output package clause: {}",
-                file_clause, pkg_clause
-            )));
-        }
+        match (out_pkg_clause, file_clause) {
+            (Some(out_name), Some(in_name)) => {
+                if out_name != in_name {
+                    return Some(Error::from(format!(
+                        r#"file's package clause: "{}" does not match package output package clause: "{}""#,
+                        in_name, out_name
+                    )));
+                }
+            },
+            (None, Some(in_name)) => {
```

Comment:
Should this and the next merge be allowed? That way we only error if the package clauses conflict.

Reply:
I guess it depends on what we think the semantics of omitting a package clause are.

I think maybe that if you omit it, you are in the package `main`.  So omitting a package clause in a file would be okay if the other file had no package clause or `package main`, but not anything else.

Thoughts?

Reply:
Where do we establish that the default package is main?

Reply:
I think we should probably hold off on this idea and the current code is good. I'm not sure the answer to any of the above questions and I think they're really good and interesting ones.

Reply:
Ok, sounds good, I'll just merge this as-is.

@fchikwekwe We have `main` declared as the default package name unceremoniously here:
https://github.com/influxdata/flux/blob/feat/use-algo-w/libflux/src/flux/ast/mod.rs#L20

Reply:
Oh, okay. If that's the case, then maybe in the future we should add a check in this case and the next one to see if the included package is main before throwing an error?

Reply:
@fchikwekwe Yes, the logic would look something like that.  I think we'd want the error message to reflect whether we are complaining about an explicit `package main` not matching a package clause in another file, or whether the clause was omitted altogether.